Source/WebCore/style/RuleData.h

```diff
@@ -36,13 +36,14 @@ enum class MatchBasedOnRuleHash : unsigned {
     ClassC
 };
 
+enum class IsBaseAppearanceStyle : bool { No, Yes };
 enum class IsStartingStyle : bool { No, Yes };
 
 class RuleData {
 public:
     static const unsigned maximumSelectorComponentCount = 8192;
 
-    RuleData(const StyleRule&, unsigned selectorIndex, unsigned selectorListIndex, unsigned position, IsStartingStyle);
+    RuleData(const StyleRule&, unsigned selectorIndex, unsigned selectorListIndex, unsigned position, IsStartingStyle, IsBaseAppearanceStyle);
```

Comment:
It would be more future proof to have an enum covering both starting-style and base-appearance. Then we don't need to keep adding arguments.